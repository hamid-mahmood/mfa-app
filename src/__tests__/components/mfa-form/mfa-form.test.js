import { render, fireEvent } from "@testing-library/react";
import MFAForm from "../../../components/mfa-form/mfa-form";

describe("MFAForm", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<MFAForm />);
    expect(baseElement).toBeTruthy();
  });
  it("should match snapshot", () => {
    const { asFragment } = render(<MFAForm />);
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});

describe("MFAForm component", () => {
  it("should call onHandleBack when the back button is clicked", () => {
    const onHandleBack = jest.fn();
    const { getByTestId } = render(
      <MFAForm
        inputs={{}}
        onHandleChange={() => {}}
        onHandleSubmit={() => {}}
        onHandleBack={onHandleBack}
      />
    );
    fireEvent.click(getByTestId("back-button"));
    expect(onHandleBack).toHaveBeenCalled();
  });
});

describe("MFAForm component submit", () => {
  it("should call onHandleSubmit when the form is submitted", () => {
    const onHandleSubmit = jest.fn();
    const { getByTestId } = render(
      <MFAForm
        inputs={{}}
        onHandleChange={() => {}}
        onHandleSubmit={onHandleSubmit}
        onHandleBack={() => {}}
      />
    );
    const form = getByTestId("add-mfa-form");
    fireEvent.submit(form);
    expect(onHandleSubmit).toHaveBeenCalled();
  });
});
