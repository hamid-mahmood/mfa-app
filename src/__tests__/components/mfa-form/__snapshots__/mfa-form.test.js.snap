// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MFAForm should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-container"
  >
    <div
      class="back-btn"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-left-long "
        data-icon="left-long"
        data-prefix="fas"
        data-testid="back-button"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M177.5 414c-8.8 3.8-19 2-26-4.6l-144-136C2.7 268.9 0 262.6 0 256s2.7-12.9 7.5-17.4l144-136c7-6.6 17.2-8.4 26-4.6s14.5 12.5 14.5 22l0 72 288 0c17.7 0 32 14.3 32 32l0 64c0 17.7-14.3 32-32 32l-288 0 0 72c0 9.6-5.7 18.2-14.5 22z"
          fill="currentColor"
        />
      </svg>
       Add Account
    </div>
    <form
      class="add-mfa-form"
      data-testid="add-mfa-form"
    >
      <div
        class="form-item"
      >
        <div>
          Enter Code given by the website
        </div>
        <input
          class="width-100-percent input-field"
          name="code"
          placeholder="i.e: sw3g lasdkas adskld asd"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-item"
      >
        <div>
          Enter Account Name
        </div>
        <input
          class="width-100-percent input-field"
          name="name"
          placeholder="Account Name"
          type="text"
          value=""
        />
      </div>
      <input
        class="submit-btn"
        type="submit"
      />
    </form>
  </div>
</DocumentFragment>
`;
